### PR TITLE
`Development:` Refactor dockerfiles to be more based on original files

### PR DESCRIPTION
**Pull request by TheZoker**
The custom image builds are [currently failing](https://github.com/ls1intum/Artemis-Local-Setup-Docker/actions/runs/2711565884). Therefore I looked at the original Dockerfiles and tried to incorporate most changes from that repo into this one. I also tried to stay as close as possible to the original files, since this allows for an easier update process in the future. 
The bamboo file was based on a older Dockerfile scheme and was heavily modified in the original repo since. I needed to change quite a bit of code, therefore it would be important to test this image thoroughly. ~Since we add some custom setup into the bamboo Dockerfile, we can't use the same base image as the other projects. The `apt-utils` package is not available within the `eclipse-temurin:11` image and the `add-apt-repository -y ppa:linuxuprising/java` command does also lead to build errors. I will further investigate that, but for now the old image should work.~

### CODE REVIEW

File: bamboo/Dockerfile [hunk 1]
```diff
@@ -1,82 +1,93 @@
-FROM adoptopenjdk:8-jdk-hotspot-focal
-LABEL maintainer="Atlassian Bamboo Team" \
-      description="Official Bamboo Server Docker Image"
+ARG BASE_IMAGE=adoptopenjdk:8-jdk-hotspot-focal
+FROM $BASE_IMAGE
 
-ENV BAMBOO_USER bamboo
-ENV BAMBOO_GROUP bamboo
+LABEL maintainer="dc-deployments@atlassian.com"
```
Comment (Mtze) [inline bamboo/Dockerfile]: I guess we should put us in here

File: jira/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-ARG BASE_IMAGE=adoptopenjdk/openjdk11
+ARG BASE_IMAGE=eclipse-temurin:11
 FROM $BASE_IMAGE
 
 LABEL maintainer="dc-deployments@atlassian.com"
```
Comment (Mtze) [inline jira/Dockerfile]: Same as above

File: bamboo/Dockerfile (hunk 1 above)
Comment (TheZoker) [reply bamboo/Dockerfile]: Done

File: jira/Dockerfile (hunk 2 above)
Comment (TheZoker) [reply jira/Dockerfile]: Done

**Review by Hialus — COMMENTED** [review]
I tested the images locally on an M1 Pro MacBook using my existing data.
BitBucket and Bamboo seem to work fine.
But Jira doesn't work correctly. While the Docker output states that Jira started, I am unable to access it in the browser:
grafik
BitBucket also outputs an error that it is unable to connect to Jira.
The only error logs I could find are at the end of this comment, but they are not very helpful IMO. If you know where I can get more detailed error reports, let me know.

The output in the Docker console (not sure if this is an error or normal for our local setup)
```
artemis_jira       | 27-Jul-2022 19:57:48.505 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["http-nio-8080"]
artemis_jira       | 27-Jul-2022 19:57:48.534 INFO [main] org.apache.catalina.startup.Catalina.start Server startup in 3872 ms
artemis_jira       | 2022-07-27 19:57:48,658+0000 JIRA-Bootstrap INFO      [c.a.j.config.database.SystemDatabaseConfigurationLoader] Reading database configuration from /var/atlassian/application-data/jira/dbconfig.xml
artemis_jira       | 2022-07-27 19:57:48,680+0000 JIRA-Bootstrap INFO      [c.a.j.config.database.DatabaseConfigHandler] Trying to get encrypted password from xml and decrypt it
artemis_jira       | 2022-07-27 19:57:48,683+0000 JIRA-Bootstrap INFO      [c.a.j.config.database.DatabaseConfigHandler] Database password decryption not performed.
artemis_jira       | 2022-07-27 19:57:49,289+0000 JIRA-Bootstrap INFO      [c.a.jira.startup.JiraStartupLogger] Running Jira startup checks.
artemis_jira       | 2022-07-27 19:57:49,310+0000 JIRA-Bootstrap FATAL      [c.a.jira.startup.JiraStartupLogger] Startup check failed. Jira will be locked.
artemis_jira       | 2022-07-27 19:57:49,363+0000 JIRA-Bootstrap INFO      [c.a.jira.startup.LauncherContextListener] Startup is complete. Jira is ready to serve.

```

The log file output when I try to access Jira in the browser:
```
2022-07-27 19:52:30,604+0000 http-nio-8080-exec-6 WARN anonymous 1192x96x1 - 172.20.0.3 /rest/usermanagement/1/event [c.a.j.config.properties.ApplicationPropertiesStore] Exception getting property 'jira.security.csp.sandbox.excluded.content.type' from database. Using default
2022-07-27 19:52:30,606+0000 http-nio-8080-exec-6 WARN anonymous 1192x96x1 - 172.20.0.3 /rest/usermanagement/1/event [c.a.j.config.properties.ApplicationPropertiesStore] Exception getting property 'jira.security.csp.sandbox.included.content.disposition' from database. Using default
2022-07-27 19:52:30,606+0000 http-nio-8080-exec-6 WARN anonymous 1192x96x1 - 172.20.0.3 /rest/usermanagement/1/event [c.a.j.config.properties.ApplicationPropertiesStore] Exception getting property 'jira.security.csp.sandbox.browser.differentiated.paths' from database. Using default
2022-07-27 19:52:30,607+0000 http-nio-8080-exec-6 WARN anonymous 1192x96x1 - 172.20.0.3 /rest/usermanagement/1/event [c.a.j.config.properties.ApplicationPropertiesStore] Exception getting property 'jira.security.csp.sandbox.browser.differentiated.clauses' from database. Using default
2022-07-27 19:52:30,610+0000 http-nio-8080-exec-6 ERROR   [o.a.c.c.C.[.[localhost].[/].[default]] Servlet.service() for servlet [default] in context with path [] threw exception
java.lang.UnsupportedOperationException: Not implemented for JIRA bootstrap time
	at com.atlassian.jira.security.login.BootstrapLoginManagerImpl.performElevatedSecurityCheck(BootstrapLoginManagerImpl.java:34)
	at com.atlassian.jira.security.login.JiraElevatedSecurityGuard.performElevatedSecurityCheck(JiraElevatedSecurityGuard.java:22)
	at com.atlassian.seraph.filter.PasswordBasedLoginFilter.runAuthentication(PasswordBasedLoginFilter.java:117)
	at com.atlassian.seraph.filter.PasswordBasedLoginFilter.login(PasswordBasedLoginFilter.java:78)
	at com.atlassian.seraph.filter.BaseLoginFilter.doFilter(BaseLoginFilter.java:110)
	... 17 filtered
	at com.atlassian.jira.servermetrics.CorrelationIdPopulatorFilter.doFilter(CorrelationIdPopulatorFilter.java:30)
	... 23 filtered
	at com.atlassian.jira.servermetrics.MetricsCollectorFilter.doFilter(MetricsCollectorFilter.java:25)
	... 25 filtered
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:829)
2022-07-27 19:52:30,613+0000 http-nio-8080-exec-6 WARN  -  [c.a.j.config.properties.ApplicationPropertiesStore] Exception getting property 'jira.security.csp.sandbox.excluded.content.type' from database. Using default
2022-07-27 19:52:30,613+0000 http-nio-8080-exec-6 WARN  -  [c.a.j.config.properties.ApplicationPropertiesStore] Exception getting property 'jira.security.csp.sandbox.included.content.disposition' from database. Using default
2022-07-27 19:52:30,614+0000 http-nio-8080-exec-6 WARN  -  [c.a.j.config.properties.ApplicationPropertiesStore] Exception getting property 'jira.security.csp.sandbox.browser.differentiated.paths' from database. Using default
2022-07-27 19:52:30,615+0000 http-nio-8080-exec-6 WARN  -  [c.a.j.config.properties.ApplicationPropertiesStore] Exception getting property 'jira.security.csp.sandbox.browser.differentiated.clauses' from database. Using default
2022-07-27 19:52:30,616+0000 http-nio-8080-exec-6 ERROR  -  [c.a.j.web.servlet.InternalServerErrorServlet] Cannot render the 500 page for error e11b98bc-6b8e-45c5-b74f-6c6248d41aa6
java.lang.UnsupportedOperationException: Not implemented for JIRA bootstrap time
	at com.atlassian.jira.security.login.BootstrapLoginManagerImpl.performElevatedSecurityCheck(BootstrapLoginManagerImpl.java:34)
	at com.atlassian.jira.security.login.JiraElevatedSecurityGuard.performElevatedSecurityCheck(JiraElevatedSecurityGuard.java:22)
	at com.atlassian.jira.security.login.JiraSeraphAuthenticator.getUserFromBasicAuthentication(JiraSeraphAuthenticator.java:119)
	at com.atlassian.seraph.auth.DefaultAuthenticator.getUser(DefaultAuthenticator.java:341)
	at com.atlassian.seraph.auth.AbstractAuthenticator.getUser(AbstractAuthenticator.java:45)
	at com.atlassian.jira.web.action.util.errors.SeraphUserRetriever.getLoggedInUser(SeraphUserRetriever.java:24)
	at com.atlassian.jira.web.servlet.InternalServerErrorHelper.<init>(InternalServerErrorHelper.java:98)
	at com.atlassian.jira.web.servlet.InternalServerErrorHelper.render500ResponsePage(InternalServerErrorHelper.java:52)
	at com.atlassian.jira.web.servlet.InternalServerErrorServlet.doRequest(InternalServerErrorServlet.java:14)
	at com.atlassian.jira.web.servlet.InternalServerErrorServlet.doGet(InternalServerErrorServlet.java:20)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:655)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
	... 51 filtered
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:829)
2022-07-27 19:52:30,623+0000 http-nio-8080-exec-7 WARN anonymous 1192x97x1 - 172.20.0.3 /rest/usermanagement/1/search [c.a.j.config.properties.ApplicationPropertiesStore] Exception getting property 'jira.security.csp.sandbox.excluded.content.type' from database. Using default
2022-07-27 19:52:30,624+0000 http-nio-8080-exec-7 WARN anonymous 1192x97x1 - 172.20.0.3 /rest/usermanagement/1/search [c.a.j.config.properties.ApplicationPropertiesStore] Exception getting property 'jira.security.csp.sandbox.included.content.disposition' from database. Using default
2022-07-27 19:52:30,625+0000 http-nio-8080-exec-7 WARN anonymous 1192x97x1 - 172.20.0.3 /rest/usermanagement/1/search [c.a.j.config.properties.ApplicationPropertiesStore] Exception getting property 'jira.security.csp.sandbox.browser.differentiated.paths' from database. Using default
2022-07-27 19:52:30,626+0000 http-nio-8080-exec-7 WARN anonymous 1192x97x1 - 172.20.0.3 /rest/usermanagement/1/search [c.a.j.config.properties.ApplicationPropertiesStore] Exception getting property 'jira.security.csp.sandbox.browser.differentiated.clauses' from database. Using default
2022-07-27 19:52:30,627+0000 http-nio-8080-exec-7 ERROR   [o.a.c.c.C.[.[localhost].[/].[default]] Servlet.service() for servlet [default] in context with path [] threw exception
java.lang.UnsupportedOperationException: Not implemented for JIRA bootstrap time
	at com.atlassian.jira.security.login.BootstrapLoginManagerImpl.performElevatedSecurityCheck(BootstrapLoginManagerImpl.java:34)
	at com.atlassian.jira.security.login.JiraElevatedSecurityGuard.performElevatedSecurityCheck(JiraElevatedSecurityGuard.java:22)
	at com.atlassian.seraph.filter.PasswordBasedLoginFilter.runAuthentication(PasswordBasedLoginFilter.java:117)
	at com.atlassian.seraph.filter.PasswordBasedLoginFilter.login(PasswordBasedLoginFilter.java:78)
	at com.atlassian.seraph.filter.BaseLoginFilter.doFilter(BaseLoginFilter.java:110)
	... 17 filtered
	at com.atlassian.jira.servermetrics.CorrelationIdPopulatorFilter.doFilter(CorrelationIdPopulatorFilter.java:30)
	... 23 filtered
	at com.atlassian.jira.servermetrics.MetricsCollectorFilter.doFilter(MetricsCollectorFilter.java:25)
	... 25 filtered
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:829)
2022-07-27 19:52:30,631+0000 http-nio-8080-exec-7 WARN  -  [c.a.j.config.properties.ApplicationPropertiesStore] Exception getting property 'jira.security.csp.sandbox.excluded.content.type' from database. Using default
2022-07-27 19:52:30,631+0000 http-nio-8080-exec-7 WARN  -  [c.a.j.config.properties.ApplicationPropertiesStore] Exception getting property 'jira.security.csp.sandbox.included.content.disposition' from database. Using default
2022-07-27 19:52:30,633+0000 http-nio-8080-exec-7 WARN  -  [c.a.j.config.properties.ApplicationPropertiesStore] Exception getting property 'jira.security.csp.sandbox.browser.differentiated.paths' from database. Using default
2022-07-27 19:52:30,633+0000 http-nio-8080-exec-7 WARN  -  [c.a.j.config.properties.ApplicationPropertiesStore] Exception getting property 'jira.security.csp.sandbox.browser.differentiated.clauses' from database. Using default
2022-07-27 19:52:30,634+0000 http-nio-8080-exec-7 ERROR  -  [c.a.j.web.servlet.InternalServerErrorServlet] Cannot render the 500 page for error e2ffe276-d5f9-4248-966f-84b0016fad3e
java.lang.UnsupportedOperationException: Not implemented for JIRA bootstrap time
	at com.atlassian.jira.security.login.BootstrapLoginManagerImpl.performElevatedSecurityCheck(BootstrapLoginManagerImpl.java:34)
	at com.atlassian.jira.security.login.JiraElevatedSecurityGuard.performElevatedSecurityCheck(JiraElevatedSecurityGuard.java:22)
	at com.atlassian.jira.security.login.JiraSeraphAuthenticator.getUserFromBasicAuthentication(JiraSeraphAuthenticator.java:119)
	at com.atlassian.seraph.auth.DefaultAuthenticator.getUser(DefaultAuthenticator.java:341)
	at com.atlassian.seraph.auth.AbstractAuthenticator.getUser(AbstractAuthenticator.java:45)
	at com.atlassian.jira.web.action.util.errors.SeraphUserRetriever.getLoggedInUser(SeraphUserRetriever.java:24)
	at com.atlassian.jira.web.servlet.InternalServerErrorHelper.<init>(InternalServerErrorHelper.java:98)
	at com.atlassian.jira.web.servlet.InternalServerErrorHelper.render500ResponsePage(InternalServerErrorHelper.java:52)
	at com.atlassian.jira.web.servlet.InternalServerErrorServlet.doRequest(InternalServerErrorServlet.java:14)
	at com.atlassian.jira.web.servlet.InternalServerErrorServlet.doPost(InternalServerErrorServlet.java:26)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:681)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:764)
	... 51 filtered
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by TheZoker** [comment]
Thanks for testing this! Can you please try the following two things:

1. Create a new clean Jira container and see if that works (so without any existing data)
2. Change the base image back to `adoptopenjdk/openjdk11`, since this is (IMHO) the only real change within this Dockerfile

Thanks!

**Comment by Mtze** [comment]
What is the current state on this PR?

**Comment by TheZoker** [comment]
I started all applications with my rebase from today and they start up without any issue:
image

image

image

**Comment by Mtze** [comment]
@Hialus Could you test this PR once again? This would be very helpful for me. Thanks in advance

**Review by Hialus — APPROVED** [review]
Tested this again on my M1 Mac. Could not spot any obvious issues.
The previous issue I encountered was caused by a very outdated Jira H2 database, which seems to have broken with some new Jira release. After installing it all from scratch it worked just fine.
Good job on the refactoring!

**Review by Mtze — APPROVED** [review]

**Pull request merged**